src/test/kotlin/org/opensearch/replication/ReplicationHelpers.kt

```diff
@@ -350,9 +352,14 @@ fun RestHighLevelClient.updateAutoFollowPattern(connection: String, patternName:
                                      }""")
     }
     lowLevelRequest.setOptions(requestOptions)
-    val lowLevelResponse = lowLevelClient.performRequest(lowLevelRequest)
-    val response = getAckResponse(lowLevelResponse)
-    assertThat(response.isAcknowledged).isTrue()
+    try {
+        val lowLevelResponse = lowLevelClient.performRequest(lowLevelRequest)
+        val response = getAckResponse(lowLevelResponse)
+        assertThat(response.isAcknowledged).isTrue()
+    } catch (e: ResponseException) {
+        // Skip if ignoreIfExists is true and exception contains resource_already_exists_exception
+        if  ((ignoreIfExists == true && e.message?.contains("resource_already_exists_exception")!!) == false) throw e
```

Comment:
If ignoreIfExists is false then exception will not be thrown. Is that the expected behavior.

Reply:
sorry, the check is slightly confusing. If ignoreIfExists is false, it will throw the exception